syntax = "proto3";
package assignment;

message Assignment {
    string public_uuid = 1;
    string instructor_public_uuid = 2;
    string version_uuid = 3;
    string rubric_version_uuid = 4;
    string name = 5;
    string description = 6;
    uint64 version_timestamp = 7;
}

message Rubric {
    string public_uuid = 1;
    string version_uuid = 2;
    string instructor_public_uuid = 3;
    string name = 4;
    map<string, RubricCriterion> rubric_criteria = 5;
    uint64 version_timestamp = 6;
}

message RubricCriterion {
    string uuid = 1;
    string name = 2;
    repeated CriterionLevel levels = 3;
}

message CriterionLevel {
    optional uint32 score = 1;
    string description = 2;
}
